products/sle/main.pm

```diff
@@ -1132,6 +1135,8 @@ sub load_patching_tests {
         # Save VERSION to UPGRADE_TARGET_VERSION
         set_var('UPGRADE_TARGET_VERSION', get_var('VERSION'));
         loadtest "migration/version_switch_origin_system";
+        # Set DM_NEEDS_USERNAME=1 for SLES11-SP4
+        set_var('DM_NEEDS_USERNAME', get_var('DM_NEEDS_USERNAME', 1)) if check_var('VERSION', '11-SP4');
```

Comment:
That's correct way, but I have a question here: should we reset DM_NEEDS_USERNAME after upgrade to SLE 15?

Reply:
Further verification test run is required to make sure it works.

Reply:
I pefer to move this back to lib/opensusebasetest.pm with check_var('VERSION', '11-SP4') instead of set and reset DM_NEEDS_USERNAME here.  "VERSION=11-SP4" is a parameter which we can make sure where it should be used  when it will be set as "11-SP4". @okurz How do you think about it?

Reply:
I have done more inverstagtion here about reset DM_NEEDS_USERNAME, if I add reset part after version_switch_upgrade_target, but the reset part will effect immdiate before login to sle11-sp4.
That is to say I need modify another pm to add reset part. (Maybe in version_switch_upgrade_target.pm)

Reply:
Yes, you have to set DM_NEEDS_USERNAME in version_switch_origin_system.pm, then reset it later in version_switch_upgrade_target.pm

It is also fine to me with check_var('VERSION', '11-SP4') in lib/opensusebasetest.pm if only one line change is required to match the specific change. But version switch would be better if you have to use check_var('VERSION', '11-SP4') several times in different modules.